ui/vic-ui-h5c/vic/src/vic-app/src/app/services/vm-view.service.ts

```diff
@@ -0,0 +1,171 @@
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+import { Subject } from 'rxjs/Subject';
+import { Http } from '@angular/http';
+
+import { VirtualContainerHost } from '../vch-view/vch.model';
+import { ContainerVm } from '../container-view/container.model';
+import { GlobalsService } from '../shared/globals.service';
+import { getVchResponseStub } from '../services/mocks/vch.response';
+import { getContainerResponseStub } from '../services/mocks/container.response';
+
+const ALL_VCHS_URL = '/ui/vic/rest/data/list/?targetType=' +
+    'vic:VirtualContainerHostVm&properties=match,results';
+const ALL_CONTAINERS_URL = '/ui/vic/rest/data/list/?targetType=' +
+    'vic:ContainerVm&properties=match,results';
+
+@Injectable()
+export class VicVmViewService {
+    private vchsSubj: Subject<VirtualContainerHost[]>;
+    private containersSubj: Subject<ContainerVm[]>;
+    public vchs$: Observable<VirtualContainerHost[]>;
+    public containers$: Observable<ContainerVm[]>;
+
+    constructor(
+        private http: Http,
+        private gs: GlobalsService
+    ) {
+        // sets up subjects and observables
+        this.vchsSubj = new Subject<VirtualContainerHost[]>();
+        this.containersSubj = new Subject<ContainerVm[]>();
+        this.vchs$ = this.vchsSubj.asObservable();
+        this.containers$ = this.containersSubj.asObservable();
+    }
+
+    /**
+     * Queries vic-service and gets a list of VCHs
+     * @param params : query parameters to query backend with
+     */
+    reloadVchs(params: {
+        offset?: number,
+        maxResultCount?: number,
+        sorting?: string,
+        filter?: string
+    }) {
+        let vchsArray: VirtualContainerHost[] = [];
+        let vchDataResponse;
+
+        // uses mocked data if app is running in standalone (dev) mode
+        if (!this.gs.isPluginMode()) {
+            vchDataResponse = getVchResponseStub().results;
+            vchsArray = [];
+
+            for (let objectId in vchDataResponse) {
+                if (vchDataResponse.hasOwnProperty(objectId)) {
+                    vchsArray.push(new VirtualContainerHost(vchDataResponse[objectId]));
+                }
+            }
+
+            this.vchsSubj.next(vchsArray);
+            return;
+        }
+
+        // makes a GET request to vic-service and emits data fetched
+        this.http.get(this.buildQueryString(ALL_VCHS_URL, params))
```

Comment:
I believe this is also a service class so it should be ok I think. Let me know if you think otherwise.